apps/api/src/app/workflows-v2/usecases/generate-preview/generate-preview.usecase.ts

```diff
@@ -141,15 +143,35 @@ export class GeneratePreviewUsecase {
     if (!step.template || !step.template.controls) {
       throw new StepMissingControlsException(command.stepDatabaseId, step);
     }
+    const origin = this.buildOrigin(persistedWorkflow);
 
     return {
       workflowId: persistedWorkflow.triggers[0].identifier,
       stepId: step.stepId,
       stepType: step.template.type,
       stepControlSchema: step.template.controls,
-      origin: persistedWorkflow.origin,
+      origin,
     };
   }
+
+  /**
+   * Builds the origin of the workflow based on the workflow type.
+   * If the origin is not set, it will be built based on the workflow type.
+   * We need to do so for backward compatibility reasons.
+   */
+  private buildOrigin(persistedWorkflow: NotificationTemplateEntity): WorkflowOriginEnum {
+    if (persistedWorkflow.origin) {
+      return persistedWorkflow.origin;
+    }
+
+    if (persistedWorkflow.type === WorkflowTypeEnum.ECHO || persistedWorkflow.type === WorkflowTypeEnum.BRIDGE) {
+      return WorkflowOriginEnum.EXTERNAL;
+    } else if (persistedWorkflow.type === WorkflowTypeEnum.REGULAR) {
```

Comment:
There is no need for else when using return ;)
```suggestion
    }
    if (persistedWorkflow.type === WorkflowTypeEnum.REGULAR) {
```

Reply:
you right, the `returns` were added after a refactor 😅